### DISCUSSION TITLE
hotfixes

### DISCUSSION THREAD

**Discussion by YottaYocta** [ArthurSonzogni/FTXUI#271] [Q&A] [answered]
I noticed that certain issues, like the one from a couple of weeks ago about the input not holding on to the reference, are added to the fix-251 branch. Since the base functionality of the input system requires the ability to access and modify the content that the users enter, could we have smaller hotfixes that get pulled into master more frequently? Certain essential features, such as the ability to clear the content of the input field on return are necessary for many applications involving input. Thanks!

**Comment by ArthurSonzogni** [comment, marked as answer]
`Input` is different from others components, because this is the only component modifying its input.

We maintain two version of every component, one supporting `std::wstring` (legacy/depreacted) and one supporting `std::string`. Fortunately the implementation are all made from `std::string` and we just convert `std::wstring` to `std::string` in constructor for legacy users.

Every components are made this way, except for `Input`, which behave in the opposite order. We convert `std::string` to `std::wstring` and back when updated. 

I believe we should make `Input` implemented in two version for both type. That requires some work.

**Reply by ArthurSonzogni** [reply]
Fixed in PR:
https://github.com/ArthurSonzogni/FTXUI/pull/279
